src/microbe_stage/organelle_components/StorageComponent.cs

```diff
@@ -29,3 +29,19 @@ public void Check(string name)
         }
     }
 }
+
+[JSONDynamicTypeAllowed]
+public class StorageComponentUpgrades : IComponentSpecificUpgrades
+{
+    public StorageComponentUpgrades(Compound? specializationFor)
+    {
+        SpecializationFor = specializationFor;
+    }
+
+    public Compound? SpecializationFor { get; set; }
```

Comment:
This doesn't sound grammatically correct to me, is it? To me `SpecializedFor` would sound grammatically correct when reading code aloud that uses this property. Would that be better?